search/src/main/java/com/jjswigut/search/ui/RestaurantListFragment.kt

```diff
@@ -0,0 +1,118 @@
+
+@AndroidEntryPoint
+class RestaurantListFragment : Fragment() {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+    private val viewModel: RestaurantListViewModel by viewModels { viewModelFactory }
+
+    private lateinit var listAdapter: RestaurantListAdapter
+    private val args: RestaurantListFragmentArgs by navArgs()
+
+    private var _binding: FragmentRestaurantlistBinding? = null
+    private val binding get() = _binding!!
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        listAdapter = RestaurantListAdapter(::handleSwipe, viewModel, requireContext())
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentRestaurantlistBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupCardStackView()
+        getRestaurants()
+        observeRestaurants()
+
+        binding.rewindButton.setOnClickListener {
+            binding.cardStack.rewind()
+
+        }
+
+
+    }
+
+    private fun handleSwipe(action: CardAction) {
+        viewModel.saveLikedRestaurants(action)
+    }
+
+    private fun handleRewind() {
+
+    }
+
+
+    private fun setupCardStackView() {
+        listAdapter.cardManager.setStackFrom(StackFrom.Top)
+        listAdapter.cardManager.setVisibleCount(5)
+        listAdapter.cardManager.setTranslationInterval(12.0f)
+        listAdapter.cardManager.setScaleInterval(0.85f)
+        listAdapter.cardManager.setSwipeThreshold(0.3f)
+        listAdapter.cardManager.setMaxDegree(20.0f)
+        listAdapter.cardManager.setDirections(Direction.HORIZONTAL)
+        listAdapter.cardManager.setCanScrollHorizontal(true)
+        listAdapter.cardManager.setCanScrollVertical(true)
+        listAdapter.cardManager.setSwipeableMethod(SwipeableMethod.AutomaticAndManual)
+        binding.cardStack.adapter = listAdapter
+        binding.cardStack.layoutManager = listAdapter.cardManager
+
+    }
+
+
+    private fun getRestaurants() {
+        viewModel.getRestaurants(args.foodType, args.radius, args.lat, args.lon)
+            .observe(viewLifecycleOwner, {
+                if (!it.data.isNullOrEmpty()) {
+                    viewModel.restaurantListLiveData.value = it.data
+                    val list = it.data
+                    if (list != null) {
+                        this.listAdapter.updateData(list)
+                    }
+
+                }
+            })
+    }
+
+    private fun observeRestaurants() {
+        viewModel.restaurantListLiveData.observe(viewLifecycleOwner, {
+            listAdapter.updateData(it)
+        })
+    }
+
+
+    override fun onDestroy() {
+        super.onDestroy()
+        _binding = null
```

Comment:
For all fragments? Isn't it in onDestroy in the documentation?

Reply:
onDestroy is a valid lifecycle callback of a fragment, but just not where you want to be nulling out your binding. your view is no longer accessible after destroyView is called, so that's where you'd want to null it out